docs/docs-beta/docs/guides/build/project-structure/creating-a-component.md

```diff
@@ -0,0 +1,120 @@
+---
+title: "Creating a New Component Type"
+sidebar_position: 100
+---
+
+:::note
+Refer to the project structuring guide to learn how to create a components-compatible project.
+:::
+
+The `dagster-components` system makes it easy to create new component types that can be reused across your project.
+
+In most cases, component types map to a specific technology. For example, you might have a `DockerScriptComponent` that executes a script in a Docker container, or a `SnowflakeQueryComponent` that runs a query on Snowflake.
+
+## Scaffolding a new component type
+
+For this example, we'll write a lightweight component that executes a shell command.
+
+First, we use the `dg` command-line utility to scaffold a new component type:
+
+```bash
+dg component-type generate shell_command
+```
+
+This will add a new file to your project in the `lib` directory:
+
+<CodeExample filePath="guides/components/shell-command/empty.py" language="python" />
+
+This file contains the basic structure for the new component type. There are two methods that you'll need to implement:
+
+- `get_schema`: This method should return a Pydantic model that defines the schema for the component. This is the schema for the data that goes into `component.yaml`.
+- `load`: This method takes the loading context and returns an instance of the component class. This is where you'll load the configuration from the `component.yaml` file.
```

Comment:
Hmmmm configuration might not be the right language. At a minimum we should probably say "component configuration" to distinguish from other forms of config in the system?